libtcmu_log.c

```diff
@@ -365,3 +392,146 @@ static struct log_buf *tcmu_log_initialize(void)
 	pthread_mutex_unlock(&g_mutex);
 	return logbuf;
 }
+
+static int append_output(log_output_func f, log_close_func c, void *data,
+				int priority, int dest, const char *name)
+	
+{
+	int ret = -1;
+	char *ndup = NULL;
+	struct log_output output;
+	if (f == NULL)
+	    return -1;
+
+	if (dest == TCMU_LOG_TO_FILE) {
+		if (name == NULL)
+			return -1;
+		ndup = strdup(name);
+		if (ndup == NULL)
+	    		return -1;
+	}
+
+	output.f = f; 
+	output.c = c; 
+	output.data = data;
+	output.priority = priority;
+	output.dest = dest;
+	output.name = ndup;
+
+	darray_append(logbuf->outputs, output);
+
+	return ret;
```

Comment:
Is this function always returning -1? I see ret = -1, but never see it get updated. I think you can just drop it and return 0 here right?